Explorer/Assets/DCL/LOD/LODAsset.cs

```diff
@@ -0,0 +1,91 @@
+using DCL.AssetsProvision;
+using DCL.Profiling;
+using ECS.StreamableLoading.AssetBundles;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Utility;
+
+namespace DCL.LOD
+{
+    public struct LODAsset : IDisposable
+    {
+        public LODKey LodKey;
+        public GameObject Root;
+        public AssetBundleData AssetBundleReference;
+        private readonly bool LoadingFailed;
+
+        private readonly LODDebugInfo LODDebugInfo;
```

Comment:
I propose you to detach `Debugging Logic` from the prod-ready ready:

- It is too tightly coupled with the prod logic
- `LODDebugInfo` is a class
- `ILODSettingsAsset` does not depend on the instance: it should not be stored in the component

Expectations:
- `LODDebugInfo` can create an overhead only if `IsColorDebuging` is true

My suggestions:
- Move `LODDebugInfo` to a separate component
- Disconnect from `LODAsset`
- Handle it in a static `Utility` class/a separate system/separate branch of the logic
   - E.g. you can move `sceneLODInfo.IsDirty = false;` to the later system and introduce a system between `UpdateSceneLODInfoSystem` and setting dirty to false that will modify the debug component based on `sceneLODInfo.CurrentLOD`

Reply:
I followed your inspiration, but did it a little bit different.

I did a system where every debug settings is encapsulated. Its a little bit expensive to turn on and to keep it running when its on due to some checks I got to do, but when tis off it has 0 effect. 

WDYT?

![image](https://github.com/decentraland/unity-explorer/assets/1999557/7b401927-a8dd-4f42-8ec1-57105532240f)
![image](https://github.com/decentraland/unity-explorer/assets/1999557/325778a9-0029-40e4-831f-5a87edb5af2a)
![image](https://github.com/decentraland/unity-explorer/assets/1999557/d6b6d99e-f702-4324-89cf-3ba2a28cab8b)

Reply:
Removed allocations alltogether